src/content/1.0.0/library/clojure/clojure-overview.md

```diff
@@ -0,0 +1,208 @@
+                             :book/tags [:rabbits :fiction]
+                             :book/authors [{:_id :author :name "Richard Adams"}]}
+                            {:_id :book
+                             :book/title "The Joy of Clojure"
+                             :book/tags [:programming :non-fiction]
+                             :book/authors [{:_id :author :name "Michael Fogus"}
+                                       {:_id :author :name "Chris Houser"}]}
+                            {:_id :book
+                             :book/title "Cryptonomicon"
+                             :book/tags [:fiction :sci-foo]
+                             :book/authors [{:_id :author :name "Neal Stephenson"}]}])
+
+@(fdb/transact conn ledger [{:_id [:book/title "Cryptonomicon"]
+                             :book/notes "I rly liked it."}])
+
+;; updating a single-cardinality predicate
+@(fdb/transact conn ledger [{:_id [:book/title "Cryptonomicon"]
+                             :book/notes "I really liked it."}])
+
+;; updating a multi-cardinality predicate
+@(fdb/transact conn ledger [{:_id [:book/title "The Joy of Clojure"]
+                             :book/title "The Joy of Clojure"
+                             :book/tags [:favorite]}])  ; adding the :favorite tag
+
+;; retracting a predicate
+@(fdb/transact conn ledger [{:_id [:book/title "Cryptonomicon"] :book/notes nil}])
+
+;; retracting multi-cardinality member
+@(fdb/transact conn ledger [{:_id [:book/title "The Joy of Clojure"]
+                             :_action :delete
+                             :book/tags [:favorite]}]) ; only removing :favorite tag
+
+;; retracting a whole subject
+@(fdb/transact conn ledger [{:_id [:book/title "Watership Down"] :_action :delete}])
+
+```
+
+### Querying
+
+#### Analytical queries
+
+The syntax for these queries is similar to datalog and eql. These don't support keywords for matching on flake values in the `:where` clause.
+
+The `query` function takes a `db` value to execute a query against, as well as a query map. See the docs for [Analytical queries](/docs/1.0.0/query/analytical-query) and [Basic queries](/docs/1.0.0/query/overview) for more details.
+
+```all
+;; all books
+@(fdb/query (db conn ledger) {:select {"?book" [:*, {:book/authors [:*]}]}
+                              :where [["?book" "book/title" nil]]})
+
+;; all non-fiction books
+@(fdb/query (db conn ledger) {:select "?title"
+                              :where [["?book" "book/title" "?title"]
+                                      ["?book" "book/tags" "non-fiction"]]})
+
+```
+
+#### History queries
+
+[History queries](/docs/1.0.0/query/history-query) can show you the raw history of a subject. Use `prettyPrint` to get back maps of predicate names to object values instead of raw flake data.
+
+```all
+;; see the history of the "Cryptonomicon" book.
+@(fdb/history-query (db conn ledger) {:history ["book/title" "Cryptonomicon"] :prettyPrint true})
```

Comment:
We could consider adding :pretty-print as the Clojure option and then in the Clojure API convert this to :prettyPrint. Same with other camel-case options. Not needed right now, but might be a good direction.

Reply:
Looking into it, it looks like `block-query` and `history-query` both have special handling for kebab casing `:pretty-print` already, so I'll update the docstrings and the example.

As a side note, should we make pretty printing the default? maybe have a `:flakes` flag instead?

Reply:
I can't recall everything pretty-print does. By the name I'd expect it to make things easier for a human but less so for a computer. I assume at least one thing is to fill in the full predicate names - but a computer will just turn around and need the predicate IDs - but I think I'd have to look at this a bit more and perhaps talk about the right behavior and ensure it is consistent across different uses.